tests/gtest/cppapi_test.cpp

```diff
@@ -264,6 +264,64 @@ TEST_F(SubvolumeTest, ManyFills)
 }
 
 
+TEST_F(SubvolumeTest, NearTraceBoundary)
+{
+    static constexpr int nrows = 3;
+    static constexpr int ncols = 2;
+    static constexpr std::size_t size = nrows * ncols;
+
+    std::array<float, size> primary_surface_data = {
+        20, 20,
+        20, 20,
+        20, 19,
+    };
+
+    std::array<float, size> top_surface_data = {
+        12, 12,
+        11, 8,
+        7, 4,
+    };
+
+    std::array<float, size> bottom_surface_data = {
+        40, 39,
+        36, 35,
+        32, 32,
+    };
+
+    const std::size_t expected_fetched_size = 10;
```

Comment:
I do not understand the purpose of this test. 
Independently of the top and bottom surfaces the entire trance is returned. 
The check consist of verifying that all 10 sample positions are in all six traces?

Reply:
Ok, then I understand. Returning all 10 samples is probably among the most likely results in case of a bug and only checking annotated position do not distinguish the different traces. 

What about checking top and bottom separately and stopping before the edge in the non checking direction.
All data values are already available in the file under the name 'samples_10_data'. Why don't we check the data value as well as the position?

Reply:
The code handle 'top' and 'bottom' separately. Given this it makes sense to have separate tests for the two cases. In case the tests fails in the future it is useful for the developer how made the change causing the failure to know if it was the top or the bottom that failed.

Reply:
If you ask for top=4, reference=20, bottom=20, you get
4, 8, 12, 16, 20, 24, 28.

You still have no clue what really failed, you just know what was tested.

But fine, if that is important to you, lets split.